cdap-api/src/main/java/co/cask/cdap/api/lineage/operation/EndPoint.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage.operation;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * EndPoint in the operation represents the data source or data sink.
```

Comment:
An EndPoint in an operation

Reply:
I would explain here that t can have a namespace, or otherwise the namespace where the program runs is used.

Reply:
added explanation